interfaces/builtin/fwupd.go

```diff
@@ -0,0 +1,229 @@
+      member={Request,Release}Name
+      peer=(name=org.freedesktop.DBus),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/DBus
+      interface=org.freedesktop.DBus
+      member=GetConnectionUnixUser
+      peer=(label=unconfined),
+
+  # Allow binding the service to the requested connection name
+  dbus (bind)
+      bus=system
+      name="org.freedesktop.fwupd",
+`)
+
+var fwupdConnectedPlugAppArmor = []byte(`
+# Description: Allow using fwupd service. Reserved because this gives
+#  privileged access to the fwupd service.
+# Usage: reserved
+
+  # DBus accesses
+  #include <abstractions/dbus-strict>
+
+  # Allow access to fwupd service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.fwupd
+      peer=(label=###SLOT_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus.Properties
+      peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var fwupdConnectedSlotAppArmor = []byte(`
+# Description: Allow firmware update from fwupd service. Reserved because this gives
+#  privileged access to the system.
+# Usage: reserved
+
+  # Allow libfwup to access efivarfs with immutable flag
+  capability linux_immutable,
+
+  # File accesses
+  # Allow write access for efivarfs filesystem
+  /sys/firmware/efi/efivars/** rw,
+
+  # Allow write access for efi firmware updater
+  /boot/efi/EFI/ubuntu/fw/** rw,
+  # Allow access from efivar library
+  @{PROC}/@{pid}/mounts r,
+  /sys/devices/**/partition r,
+  /dev/sd[a-z]* r,
+  # Allow access UEFI firmware platform size
+  /sys/firmware/efi/ r,
+  /sys/firmware/efi/fw_platform_size r,
+
+  # Allow traffic to/from org.freedesktop.DBus for fwupd service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus.**
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/fwupd{,/**}
+      interface=org.freedesktop.DBus.**
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  # Allow traffic to/from fwupd interface with any method
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.fwupd
+      peer=(label=###PLUG_SECURITY_TAGS###),
+
+  dbus (receive, send)
+      bus=system
+      path=/org/freedesktop/fwupd{,/**}
+      interface=org.freedesktop.fwupd
+      peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var fwupdPermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="org.freedesktop.fwupd"/>
+</policy>
+<policy context="default">
+    <allow send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.fwupd"/>
+    <allow send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.DBus.Properties"/>
+    <allow send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.DBus.Introspectable"/>
+    <allow send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.DBus.Peer"/>
+</policy>
+`)
```

Comment:
Some bus policy will be needed so dbus-daemon will allow the connection and this is pretty bare bones. I prefer the way that location-control did it though-- put the root policy in PermanentSlotDBus and put the default policy in ConnectedPlugDBus.

That said, I have questions and observations:
- context=default should have `<deny own="org.freedesktop.fwupd"/>
- Are org.freedesktop.DBus.Properties, org.freedesktop.DBus.Introspectable and org.freedesktop.DBus.Peer all needed here or is org.freedesktop.fwupd enough?
- shouldn't user="root" have `<allow send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.fwupd"/>` (and the other 3 if they are required)?
- should non-root be able to talk to fwupd at all? If not, a `<deny send_destination="org.freedesktop.fwupd" send_interface="org.freedesktop.fwupd"/>` might be nice here